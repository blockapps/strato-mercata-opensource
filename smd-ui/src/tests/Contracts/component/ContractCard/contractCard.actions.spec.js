import {
  fetchState,
  fetchStateSuccess,
  fetchStateFailure,
  selectContractInstance,
  fetchCirrusInstances,
  fetchCirrusInstancesSuccess,
  fetchCirrusInstancesFailure,
  fetchAccount,
  fetchAccountSuccess,
  fetchAccountFailure,
  fetchContractInfoRequest,
  fetchContractInfoSuccess,
  fetchContractInfoFailure,
} from '../../../../components/Contracts/components/ContractCard/contractCard.actions';
import { contract, modals } from './contractCardMock';

describe('ContractCard: action', () => {

  describe('fetch state', () => {

    test('request', () => {
      expect(fetchState(contract.name, contract.address, contract.chainId)).toMatchSnapshot();
    });

    test('success', () => {
      expect(fetchStateSuccess(contract.name, contract.address, contract.state)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(fetchStateFailure(contract.error)).toMatchSnapshot();
    });

  })

  test('select contract instance', () => {
    expect(selectContractInstance(contract.name, contract.address)).toMatchSnapshot();
  });

  describe('fetch instance', () => {

    test('request', () => {
      expect(fetchCirrusInstances(contract.name)).toMatchSnapshot();
    });

    test('success', () => {
      expect(fetchCirrusInstancesSuccess(contract.name, contract.instances)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(fetchCirrusInstancesFailure(contract.name, contract.error)).toMatchSnapshot();
    });

  })

  describe('fetch account', () => {

    test('request', () => {
      expect(fetchAccount(contract.name, contract.address)).toMatchSnapshot();
    });

    test('success', () => {
      expect(fetchAccountSuccess(contract.name, contract.address, contract.account)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(fetchAccountFailure(contract.name, contract.address, contract.error)).toMatchSnapshot();
    });

  })
  describe('fetch contract info snapshots', () => {

    test('request', () => {
      expect(fetchContractInfoRequest(modals.key, modals.name, modals.address, modals.chainId)).toMatchSnapshot();
    });

    test('success', () => {
      expect(fetchContractInfoSuccess(modals.key, {address: modals.address, chainId: modals.chainId, xabi: {}})).toMatchSnapshot();
    });

    test('failure', () => {
      expect(fetchContractInfoFailure(modals.key, modals.error)).toMatchSnapshot();
    });

  })
});