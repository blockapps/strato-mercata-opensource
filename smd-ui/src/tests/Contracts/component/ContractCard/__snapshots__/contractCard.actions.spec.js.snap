// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractCard: action fetch account failure 1`] = `
Object {
  "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
  "error": "ERROR",
  "name": "Greeter",
  "type": "FETCH_ACCOUNT_FAILURE",
}
`;

exports[`ContractCard: action fetch account request 1`] = `
Object {
  "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
  "name": "Greeter",
  "type": "FETCH_ACCOUNT_REQUEST",
}
`;

exports[`ContractCard: action fetch account success 1`] = `
Object {
  "account": Array [
    Object {
      "contractRoot": "1578e5fa942f475f407b9f9c67c4474bd4856a5f1935ee3fb7ffe4333018f1d7",
      "nonce": 0,
    },
  ],
  "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
  "name": "Greeter",
  "type": "FETCH_ACCOUNT_SUCCESS",
}
`;

exports[`ContractCard: action fetch contract info snapshots failure 1`] = `
Object {
  "error": Object {
    "message": "some error",
  },
  "key": "data-card-1234567-",
  "type": "FETCH_CONTRACT_INFO_FAILURE",
}
`;

exports[`ContractCard: action fetch contract info snapshots request 1`] = `
Object {
  "chainId": "123456756789876789876543345678",
  "contractAddress": "1234567",
  "contractName": "Foo",
  "key": "data-card-1234567-",
  "type": "FETCH_CONTRACT_INFO_REQUEST",
}
`;

exports[`ContractCard: action fetch contract info snapshots success 1`] = `
Object {
  "data": Object {
    "address": "1234567",
    "chainId": "123456756789876789876543345678",
    "xabi": Object {},
  },
  "key": "data-card-1234567-",
  "type": "FETCH_CONTRACT_INFO_SUCCESS",
}
`;

exports[`ContractCard: action fetch instance failure 1`] = `
Object {
  "error": "ERROR",
  "name": "Greeter",
  "type": "FETCH_CIRRUS_INSTANCES_FAILURE",
}
`;

exports[`ContractCard: action fetch instance request 1`] = `
Object {
  "chainId": undefined,
  "name": "Greeter",
  "type": "FETCH_CIRRUS_INSTANCES_REQUEST",
}
`;

exports[`ContractCard: action fetch instance success 1`] = `
Object {
  "instances": Array [
    Object {
      "address": "b7b986bf23faebd8d745c65fa42a8c2f0fc2ebb9",
      "greeting": "",
    },
  ],
  "name": "Greeter",
  "type": "FETCH_CIRRUS_INSTANCES_SUCCESS",
}
`;

exports[`ContractCard: action fetch state failure 1`] = `
Object {
  "error": "ERROR",
  "type": "FETCH_STATE_FAILURE",
}
`;

exports[`ContractCard: action fetch state request 1`] = `
Object {
  "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
  "chainId": "ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9",
  "name": "Greeter",
  "type": "FETCH_STATE_REQUEST",
}
`;

exports[`ContractCard: action fetch state success 1`] = `
Object {
  "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
  "name": "Greeter",
  "state": Object {
    "dna": "",
    "geneticallyModify": "function() {}",
    "name": "",
  },
  "type": "FETCH_STATE_SUCCESS",
}
`;

exports[`ContractCard: action select contract instance 1`] = `
Object {
  "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
  "name": "Greeter",
  "type": "SELECT_CONTRACT_INSTANCE",
}
`;
